### PR TITLE
Added WithRecipientLookup option

### PR DESCRIPTION
**Pull request by scruzin**
This makes it possible to customize notification recipients by site key and/or notification kind.

### CODE REVIEW

**Review by saxon-milton — APPROVED** [review]
One minor comment, otherwise LGTM

**Comment by scruzin** [comment]
> One minor comment, otherwise LGTM

I can't see any comments. Did you save it?

File: notify/notify_test.go [hunk 1]
```diff
@@ -167,3 +168,26 @@ func (ts *testStore) Sent(ctx context.Context, skey int64, key string) error {
...
+	if want != testRecipient {
+		t.Errorf("Recipients returned %s, expected %s", want, testRecipient)
+	}
+}
+
+// testLookup is our recipient lookup function.
+func testLookup(skey int64, kind Kind) []string {
+	if kind == "test" {
+		return []string{testRecipient}
+	} else {
```
Comment (saxon-milton) [inline notify/notify_test.go:190]: No need for else

**Comment by saxon-milton** [comment]
> > One minor comment, otherwise LGTM
> 
> I can't see any comments. Did you save it?

I did not; it's there now :)

**Pull request merged**